zigpy/zcl/clusters/greenpower.py

```diff
@@ -0,0 +1,407 @@
+"""Zigbee Green Power Domain"""
+
+from __future__ import annotations
+
+from typing import Final
+
+import zigpy.types as t
+from zigpy.types.struct import StructField
+from zigpy.zcl import Cluster, foundation
+from zigpy.zcl.foundation import (
+    BaseAttributeDefs,
+    BaseCommandDefs,
+    ZCLAttributeDef,
+    ZCLCommandDef,
+)
+import zigpy.zgp.types as zgptypes
+
+
+# Figure 27
+class CommissioningNotificationSchema(foundation.CommandSchema):
+    options: t.bitmap16
+    gpd_id: zgptypes.DeviceID
+    frame_counter: t.uint32_t
+    command_id: t.uint8_t
+    payload: t.LVBytes
+    gpp_short_addr: t.uint16_t = StructField(
+        requires=lambda s: s.proxy_info_present, optional=True
+    )
+    distance: t.uint8_t = StructField(
+        requires=lambda s: s.proxy_info_present, optional=True
+    )
+    mic: t.uint32_t = StructField(requires=lambda s: s.security_failed, optional=True)
+
+    @property
```

Comment:
Oh right, I did have a specific question: is the use of properties like this to access the options bitfield appropriate, or should I be laying them out with `reserved` padding in the Struct itself?